File: leetcode/2566-maximum-difference-by-remapping-a-digit/solution.cpp
class Solution {
public:
    int convert(int num, int from, int to){
        string str = to_string(num);
        for(int i=0; i<str.size(); i++){
            if(str[i] == from+'0') str[i] = to+'0';
        }
        return stoi(str);
    }
    int minMaxDifference(int num) {
        
        int mx = num, mn = num;
        for(int i=0; i<=9; i++){
            for(int j=0; j<=9; j++){
                int ret = convert(num, i, j);
                mx = max(mx, ret);
                mn = min(mn, ret);
            }
        }
        
        return mx-mn;
        
    }
};

File: leetcode/2566-maximum-difference-by-remapping-a-digit/solution.py
class Solution:
    def minMaxDifference(self, num: int) -> int:
        num = str(num)
        maxDigit = next((digit for digit in num if digit != '9'), '-1')
        minDigit = num[0]

        maxNum = ''.join(digit if digit != maxDigit else '9' for digit in num)
        minNum = ''.join(digit if digit != minDigit else '0' for digit in num)

        return int(maxNum) - int(minNum)
